allennlp/nn/beam_search.py

```diff
@@ -227,6 +233,12 @@ def search(self,
                         gather(1, expanded_backpointer).\
                         reshape(batch_size * self.beam_size, *last_dims)
 
+        if not torch.isfinite(last_log_probabilities).all():
+            warnings.warn("Infinite log probabilities encountered. Some final sequences may not make sense. "
+                          "This can happen when the beam size is larger than the number of valid (non-zero "
+                          "probability) transitions that the step function produces.",
+                          RuntimeWarning)
```

Comment:
Hmm, this is a bit unsatisfying, but I guess filtering out the impossible results thereby changing the shape of the returned tensor might be a worse fix. Could you add to the docstring that it's the responsibility of the caller to check the returned probabilities for invalid results?

Reply:
✅ added to the docstring